### PR TITLE
[MI-2589]: Fixed the bug of updating the redux state after login and refactor the code for redux of Jaas 

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const GetConfig = (): JSX.Element => {
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        dispatch(loadConfig());
+    }, []);
+
+    // This container is used just for making the API call for fetching the config, it doesn't render anything.
+    return <></>;
+};
```
Comment (raghavaggarwal2308) [inline webapp/src/app.tsx]:
```suggestion
const GetConfig = (): React.ReactNode => {
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(loadConfig());
    }, []);

    // This container is used just for making the API call for fetching the config, it doesn't render anything.
    return null;
};
```
try this once

File: webapp/src/jaas/components/conference/jaas_conference.tsx [hunk 2]
```diff
@@ -1,13 +1,9 @@
 import React from 'react';
-import {id as pluginId} from '../../../manifest';
-import {StartMeetingWindowAction} from 'jaas/actions';
+import {Client4} from 'mattermost-redux/client';
+import {id, id as pluginId} from '../../../manifest';
```
Comment (raghavaggarwal2308) [inline webapp/src/jaas/components/conference/jaas_conference.tsx]:
```suggestion
import {id as pluginId} from '../../../manifest';
```

File: webapp/src/jaas/components/conference/jaas_conference.tsx [hunk 3]
```diff
@@ -40,6 +36,28 @@ export default class JaaSConference extends React.PureComponent<Props, State> {
...
+            method: 'post',
+            body: JSON.stringify({jaasJwt: jwt ?? '', jaasPath: meetingId}),
+            headers: {}
+        };
+        const baseUrl = `/plugins/${id}`;
+        fetch(`${baseUrl}/api/v1/meetings/jaas/settings`, Client4.getOptions(options)).
+            then((result) => {
+                if (result.ok) {
+                    return result.json();
+                }
```
Comment (raghavaggarwal2308) [inline webapp/src/jaas/components/conference/jaas_conference.tsx:51]:
```suggestion
                }
                
```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/app.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import {loadConfig} from '../src/actions';
+import React, {useEffect} from 'react';
+import {useDispatch} from 'react-redux';
```
Comment (ayusht2810) [inline webapp/src/app.tsx:2]: write `react` imports first

File: webapp/src/jaas/components/conference/jaas_conference.tsx [hunk 5]
```diff
@@ -40,6 +36,28 @@ export default class JaaSConference extends React.PureComponent<Props, State> {
         this.api = new (window as any).JitsiMeetExternalAPI(JAAS_URL, options);
     }
 
+    startJaaSMeetingWindow(jwt: string | null, meetingId: string | null) {
+        const options = {
+            method: 'post',
+            body: JSON.stringify({jaasJwt: jwt ?? '', jaasPath: meetingId}),
+            headers: {}
+        };
+        const baseUrl = `/plugins/${id}`;
```
Comment (ayusht2810) [inline webapp/src/jaas/components/conference/jaas_conference.tsx]:
```suggestion
        };
        
        const baseUrl = `/plugins/${id}`;
```

File: webapp/src/jaas/index.tsx [hunk 6]
```diff
@@ -2,21 +2,9 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import {Provider} from 'react-redux';
 
-import store from './reducers/store';
-import JaaSConference from './components/conference';
-
+import JaaSConference from './components/conference/jaas_conference';
```
Comment (ayusht2810) [inline webapp/src/jaas/index.tsx:4]:
```suggestion
import JaaSConference from './components/conference/jaas_conference';

```

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/app.tsx (hunk 1 above)
Comment (Kshitij-Katiyar) [reply webapp/src/app.tsx]: copied this from service now and azure .Is there any difference between the two ?

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin.go [hunk 7]
```diff
@@ -729,6 +729,8 @@ func (p *Plugin) getUserConfig(userID string) (*UserConfig, error) {
 		return nil, err
 	}
 
+	userConfig.UseJaas = p.getConfiguration().UseJaaS
+
```
Comment (manojmalik20) [inline server/plugin.go]:
```suggestion
```

File: webapp/src/app.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import {loadConfig} from '../src/actions';
```
Comment (manojmalik20) [inline webapp/src/app.tsx]:
```suggestion
import {loadConfig} from './actions';
```

File: webapp/src/app.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+const GetConfig = (): React.ReactNode => {
```
Comment (manojmalik20) [inline webapp/src/app.tsx]:
```suggestion
const App = (): React.ReactNode => {
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**